app/controllers/products_common_controller.rb

```diff
@@ -77,11 +77,16 @@ def show
 
     # is the user approved?
     if add_to_cart && !@product.can_be_used_by?(acting_user) && !session_user_can_override_restrictions?(@product)
-      if TrainingRequest.submitted?(session_user, @product)
-        flash[:notice] = t_model_error(Product, "already_requested_access", product: @product)
-        return redirect_to facility_path(current_facility)
+      if SettingsHelper.feature_on?(:training_requests)
```

Comment:
Is there a way we can share this code with the instruments controller? It looks like it's the same other than the `@instrument` vs `@product`